packages/twenty-server/src/engine/core-modules/function/function.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+
+import { FileUpload } from 'graphql-upload';
+
+import { User } from 'src/engine/core-modules/user/user.entity';
+import { UserService } from 'src/engine/core-modules/user/services/user.service';
+import { InjectWorkspaceRepository } from 'src/engine/twenty-orm/decorators/inject-workspace-repository.decorator';
+import { WorkspaceRepository } from 'src/engine/twenty-orm/repository/workspace.repository';
+import {
+  FunctionSyncStatus,
+  FunctionWorkspaceEntity,
+} from 'src/modules/function/standard-objects/function.workspace-entity';
+import { CustomCodeEngineService } from 'src/engine/integrations/custom-code-engine/custom-code-engine.service';
+
+@Injectable()
+export class FunctionService {
+  constructor(
+    private readonly customCodeEngineService: CustomCodeEngineService,
+    private readonly userService: UserService,
+    @InjectWorkspaceRepository(FunctionWorkspaceEntity)
+    private readonly functionRepository: WorkspaceRepository<FunctionWorkspaceEntity>,
```

Comment:
I would store functions in metadata. As the rule of thumb, as soon as we deal with metadataIds, I would put it in metadata:
- objects, fields, workflows, workflowNodes, functions, views, ... would go there as workflow will reference workflowNodeMetadataId, and nodes will reference functionMetadataId

I think this would be a more consistent / logical approach

the draw back is that we are losing:
- schema isolation which is a great privacy feature (it's likely that some metadata rows will contain data, ex: viewFilter will contain { eq: dataValue } <== if this becomes an issue we can replace this by pointers.
- the possibility to treat these as first class citizen (aka standard objects) and to create index table of functions in the product as we do for companies. <== already discussed with product and we don't care about this one